### DISCUSSION TITLE
Storage 저장관련하여 질문드립니다.

### DISCUSSION THREAD

**Discussion by VioletOZ** [naverz/zepeto-studio-kor#112] [5. Q&A]
현재 제페토 서버에서 DB 저장이 안되어서 질문드립니다.

playerDB : DataStorage = client.loadDataStorage() ;
let jsonData = await playerDB.get('Item') as string;

ItemData = JSON.parse(jsonData);

ItemData.Inventory = 1000;

jsonData = JSON.stringfy(ItemData);

let result : boolean = await playerDB.set('Item', jsonData);

console.log(`${result}');

와 같은 식으로 하였는데
result 가 fail이 나옵니다.

별다른 이상은 없어보이는데 DB에 저장이 안되는거같습니다.

도움부탁드립니다.

**Comment by hyeonjin-jo** [comment]
안녕하세요

답변이 늦어서 죄송합니다.

내용 확인해 보니 데이터 저장은 하고 있는데 리턴값을 잘못 넘겨 주고 있는 부분을 발견해서 다음 버전에 수정하도록 하겠습니다.

감사합니다.

**Reply by VioletOZ** [reply]
넵 감사합니다.

그럼 혹시 지금 데이터 저장이 서버에서 휘발성으로 저장이 되는건가요?
서버 재가동하거나하면 이전 데이터들은 사라지는거같은데

서버가 재가동되어도 저장할 방법이 있을까요?
데이터 유지관리 하는 방법이있나요?

**Reply by hyeonjin-jo** [reply]
로컬 서버의 경우 메모리에만 데이터를 저장해서 재실행시 유지가 되지 않고 실제 앱등록 시에는 DB에 등록되어서 데이터가 유지가 되는 방식입니다.

이 부분 가이드 추가가 필요할 것 같아서 추후 반영하도록 하겠습니다.

**Reply by VioletOZ** [reply]
네 감사합니다
